### PR TITLE
Use lsp-protcol and plists

### PR DESCRIPTION
**Pull request by ericdallo**
https://github.com/emacs-lsp/lsp-mode/issues/1751

### CODE REVIEW

File: lsp-treemacs.el [hunk 1]
```diff
@@ -142,8 +142,7 @@
   (interactive)
   (let ((key (button-get (treemacs-node-at-point) :data)))
     (if (and (consp key) (ht? (cdr key)))
```
Comment (yyoncho) [inline lsp-treemacs.el:144]: this ht? should go away.

Comment (ericdallo) [reply lsp-treemacs.el:144]: thanks but the commit is WIP, I did not change every place yet :p

Comment (yyoncho) [reply lsp-treemacs.el:144]: I saw that it is wip after I did the comment.

Comment (ericdallo) [reply lsp-treemacs.el:144]: I finished the refactors, but I think this ht is for internal uses and comes from `(lsp-diagnostics)` that returns a a hash-table on `lsp-mode.el`.

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**